pkg/build/docker/docker.go

```diff
@@ -30,6 +31,20 @@ func NewBuilder(cxt *portercontext.Context) *Builder {
 }
 
 func (b *Builder) BuildInvocationImage(ctx context.Context, manifest *manifest.Manifest) error {
+	buildArgs := make(map[string]*string)
```

Comment:
Since we have this same logic in a couple places, how about we put this in a new package, pkg/build/driverutil, and then call it something like `driverutil.GetBuildArgs`

Reply:
The docker one expects a `map[string]*string` for its image's buildArgs while buildkit expects a `map[string]string`. can we achieve this with different expected output type? Should we use the build driver as a parameter and let the function determine the type of the map value from it? Example as follows.

```
func GetBuildArgs(custom manifest.CustomDefinitions, buildDriver string) (interface{}, error) {
   switch buildDriver {
    case "docker":
           output := make(map[string[*string)
           // flatten customDefinitions and put it in the map
           return output, nil
    case "buildkit"
           output := make(map[string[string)
           // flatten customDefinitions and put it in the map
           return output, nil
    default:
         return nil, errors.New("build driver is not expected: %+v", buildDriver)
    }
}
```

Reply:
What you have in the PR now is fine 👍